backend/src/repositories/BlocksSummariesRepository.ts

```diff
@@ -17,14 +17,24 @@ class BlocksSummariesRepository {
     return undefined;
   }
 
-  public async $saveSummary(height: number, summary: BlockSummary) {
+  public async $saveSummary(height: number, mined: BlockSummary | null = null, template: BlockSummary | null = null) {
```

Comment:
@nymkappa consider simplifying this a bit with a small refactor so you don't have to pass around null values which can be confusing:

```
async function $saveSummary(params : {height: number, mined?: BlockSummary, template?: BlockSummary}) {
     const { height, mined, template } = params;
...
if (mined) {
}

if (template) {
}
```

Then call it with the parameters you will actually use:
```
await BlocksSummariesRepository.$saveSummary({height: 123, mined: minedBlockSummary});
await BlocksSummariesRepository.$saveSummary({height: 123, template: templateBlockSummary})
```